### PR TITLE
Add rmath-julia

### PR DESCRIPTION
**Pull request by ararslan**
This uses JuliaLang/Rmath-julia to build and install libRmath-julia.dylib. Rmath used to be included with Julia but has been removed in 0.5. It isn't an actual *Julia* dependency; nothing in Base uses it. However, it's basically the lifeblood of the JuliaStats packages at the moment.

Please let me know if I did everything right or if there's anything I need to change, @staticfloat!

### CODE REVIEW

**Comment by tkelman** [comment]
The tap names are a bit confusing, homebrew-julia is for building a homebrew version of julia and its dependencies, homebrew-juliadeps is for use by Homebrew.jl and packages with binary dependencies (even in the non-homebrew build of Julia).

**Comment by ararslan** [comment]
> packages with binary dependencies

Wouldn't Rmath fall under that category since it's used by stats packages?

**Comment by tkelman** [comment]
Yes, I was mostly responding to the "It isn't an actual Julia dependency; nothing in Base uses it." part which I thought you were referring to w.r.t. homebrew organization. Nevermind.

**Comment by ararslan** [comment]
¯\\\_(ツ)\_/¯ I dunno what I was referring to.

**Comment by staticfloat** [comment]
Looks good to me!  Let's bottle this sucker.  Thanks @ararslan.

**Comment by ararslan** [comment]
Glad I could help! 🍻

File: rmath-julia.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  desc "The standalone Rmath library patched to use the dSFMT RNG"
+  homepage "https://github.com/JuliaLang/Rmath-julia"
+  url "https://github.com/JuliaLang/Rmath-julia/archive/v0.1.tar.gz"
+  version "0.1"
+  sha256 "8ba59378273dea999f956cecf37d47723682f3a46b55ce76ec36a0ffdbe6c8f7"
+  head "https://github.com/JuliaLang/Rmath-julia.git", :branch => "master"
+
+  option "with-clang", "Compile using Clang instead of GCC"
+
+  depends_on :gcc => :build unless "with-clang"
```
Comment (tkelman) [inline rmath-julia.rb]: does this prefer gcc? any reason for doing that?

Comment (staticfloat) [inline rmath-julia.rb]: For some reason I saw this and thought "Oh yeah, we do prefer GCC on OSX don't we", until I remembered that no, in fact, it's actually quite difficult to build Julia with GCC on OSX.  @ararslan feel free to test out the `rmath-julia` that is now live in this tap and make sure it works for you, but I am curious as to why you chose this compiler configuration.  Note that on OSX, [`USECLANG` is the default](https://github.com/JuliaLang/julia/blob/9b8e9d45e2264610db6cd057fc4747ae8a6b6249/Make.inc#L358-L359) on everything above Snow Leopard.

Comment (ararslan) [inline rmath-julia.rb]: GCC is the default in the Rmath-julia Makefile; using Clang requires a separate option to be set, so I figured I'd let it use its default behavior.

Comment (staticfloat) [inline rmath-julia.rb]: Ah, yes.  That is true in general, except on OSX.  Note that when building Julia, we always override things like `CC` and `USECLANG` when building Rmath.  I will make the requisite changes to make `clang` the default here, rebottle, and let you test again.

Comment (ararslan) [inline rmath-julia.rb]: Sorry, I didn't realize that! Thanks for making the changes. Let me know when it's ready and I'll test.

Comment (tkelman) [inline rmath-julia.rb]: >  Note that when building Julia, we always override things like CC and USECLANG when building Rmath.

used to. now that makefile isn't in the julia repo at all

Comment (staticfloat) [inline rmath-julia.rb]: Ready for testing.

Comment (ararslan) [inline rmath-julia.rb]: Works for me!

Comment (tkelman) [inline rmath-julia.rb]: sounds like i don't need to worry about building a dylib then. i guess homebrew will be used for enough other packages that amortized, it's not too bad of a dependency

Comment (ararslan) [inline rmath-julia.rb]: > sounds like i don't need to worry about building a dylib then

Hopefully that means my secret plan to try to make the Rmath transition less painful for you has worked. 😄

Comment (tkelman) [inline rmath-julia.rb]: secret plan? you've been a huge help. if you have suggestions for what we did right with onboarding to get you contributing, and what we might be doing wrong that we should adjust, would always be good to get that kind of feedback.

Comment (ararslan) [inline rmath-julia.rb]: What kind of onboarding do you mean? I just like to help. ¯\\\_(ツ)\_/¯

**Pull request merged**